ambry-test-utils/src/main/java/com/github/ambry/clustermap/MockDataNodeId.java

```diff
@@ -58,7 +58,19 @@ public MockDataNodeId(String hostname, List<Port> ports, List<String> mountPaths
    * @param dataCenter Name of datacenter.
    */
   public MockDataNodeId(List<Port> ports, List<String> mountPaths, String dataCenter) {
-    this("localhost", ports, mountPaths, dataCenter);
+    this(ports, mountPaths, dataCenter, false);
+  }
+
+  /**
+   * Create a {@link MockDataNodeId} object from given {@code ports}, {@code mountPaths}, {@code dataCenter}.
+   * @param ports Ports associated with server on node.
+   * @param mountPaths Mount paths for replicas on node.
+   * @param dataCenter Name of datacenter.
+   * @param uniqueName true if want to have unique host name. false, host name is the same.
+   */
+  public MockDataNodeId(List<Port> ports, List<String> mountPaths, String dataCenter, boolean uniqueName) {
+    this(uniqueName? "localhost" + count : "localhost", ports, mountPaths, dataCenter);
```

Comment:
why not just always set uniqueName to true?

Reply:
Yes, first I tried to always generate unique host name.
But int-test kept falling one after another like FrontendQuotaIntegrationTest. I guess quite some unit test assume the host name is always "localhost".
It may take some effort to understand these integration test cases and clean them up.
So I revert back the change and keep the original way.

Reply:
This is not the right way to do it. There are two different ways moving forward.
1. Adding port to the ReplicateBlobRequest (sorry I was against it at the first place, but not it seems like it's necessary). The port would be part of the key to identity the data node, not the port to connect to. For example, if a data node has plaintext port 12001, and ssl port 13001 and http2 port 14001. Then the port number in ReplicateBlobRequest should be 12001, since it will be used to identify the data node. But we still would connect to http2 port.
2. Fix the FrontendQuotaIntegrationTest.

If it's easier to do 1, please go for it. Sorry I misled you at the first place.